packages/bus/src/broadcast/index.test.ts

```diff
@@ -31,6 +32,93 @@ describe('broadcast', () => {
     resetMockBroadcastChannel();
   });
 
+  it('sets channel name from config', () => {
+    const bus = new Broadcast({ channelName: 'custom-channel' });
+    // @ts-expect-error Accessing protected property
+    expect(bus.channelName).toBe('custom-channel');
+  });
+
+  it('onmessage', () => {
+    const bus = Broadcast.getInstance<Payload<'test', string>>();
+    const fn = vi.fn();
+    // @ts-expect-error Accessing protected property
+    bus.handleListeners = fn;
+
+    // Simulate an error event
+    const event = new MessageEvent('message', {
+      data: 'test',
+    });
+
+    // @ts-expect-error Accessing protected property
+    bus.channel.onmessage(event);
+
+    expect(fn).toHaveBeenCalled();
+  });
+
+  it('onmessageerror', () => {
+    const bus = Broadcast.getInstance<Payload<'test', string>>();
+    const consoleMock = vi.spyOn(console, 'error');
+
+    // Simulate an error event
+    const event = new MessageEvent('message', {
+      data: 'test',
+    });
+
+    // @ts-expect-error Accessing protected property
+    bus.channel.onmessageerror(event);
+
+    expect(consoleMock).toHaveBeenCalled();
+    expect(consoleMock).toHaveBeenCalledWith(
+      'BroadcastChannel message error',
+      event,
+    );
+  });
+
+  it('setEventEmitOptions', () => {
+    const bus = Broadcast.getInstance<Payload<'test', string>>();
+
+    bus.setEventEmitOptions('test', { target: 'self' });
+    expect(
+      // @ts-expect-error Accessing protected property
+      bus.emitOptions.get('test'),
+    ).toEqual({ target: 'self' });
+
+    bus.setEventEmitOptions('test', null);
+    expect(
+      // @ts-expect-error Accessing protected property
+      bus.emitOptions.get('test'),
+    ).toBeUndefined();
+  });
+
+  it('setEventsEmitOptions', () => {
+    const bus = Broadcast.getInstance<Payload<'test', string>>();
+
+    bus.setEventsEmitOptions(new Map([['test', { target: 'self' }]]));
+    expect(
+      // @ts-expect-error Accessing protected property
+      bus.emitOptions.get('test'),
+    ).toEqual({ target: 'self' });
+  });
+
+  it('setGlobalEmitOptions', () => {
+    const bus = Broadcast.getInstance<Payload<'test', string>>();
+
+    bus.setGlobalEmitOptions({ target: 'self' });
+    expect(
+      // @ts-expect-error Accessing protected property
+      bus.emitOptions.get(bus.id),
+    ).toEqual({ target: 'self' });
+  });
+
+  it('should warn if no listeners for event', () => {
+    const bus = Broadcast.getInstance<Payload<'test', string>>();
+    const consoleMock = vi.spyOn(console, 'warn');
+
+    bus.emit('test', 'test');
+
+    expect(consoleMock).toHaveBeenCalled();
+  });
+
```

Comment:
would be good to add a test for non-broadcasts across channels if that's meant (q from my other comment:


```suggestion
it('should not postMessage when no local listeners are registered', () => {
    const bus = Broadcast.getInstance<Payload<'test', string>>();
    const consoleMock = vi.spyOn(console, 'warn');

    // No listeners registered locally - emit should warn and skip broadcasting
    bus.emit('test', 'message');

    expect(consoleMock).toHaveBeenCalledWith(
      'No listeners registered for this event type:',
      'test',
    );

    // postMessage is not called since no local listeners exist
    // @ts-expect-error Accessing protected property
    expect(bus.channel.postMessage).not.toHaveBeenCalled();
  });
```